### ISSUE TITLE
Compile errors with lftp-4.6.3a on legacy system (Redhat 9)(Kernel: 2.4.20-6)

### ISSUE THREAD

**Issue by areis422** [lavv17/lftp#171]
Receiving the following errors when compiling lftp-4.6.3a on legacy Redhat 9 server:
```
make[1]: Entering directory `/root/lftp-4.6.3a/src'
/bin/sh ../libtool --silent  --tag=CXX   --mode=link g++  -O2 -Wall -Wwrite-strings -Woverloaded-virtual -fno-exceptions -fno-rtti -fno-implement-inlines -module -avoid-version -rpath /usr/lib/lftp/4.6.3a -s -o cmd-mirror.la  MirrorJob.lo  -lutil -lcurses -lresolv -ldl
/bin/sh ../libtool --silent  --tag=CXX   --mode=link g++  -O2 -Wall -Wwrite-strings -Woverloaded-virtual -fno-exceptions -fno-rtti -fno-implement-inlines -module -avoid-version -rpath /usr/lib/lftp/4.6.3a -s -o cmd-sleep.la  SleepJob.lo  -lutil -lcurses -lresolv -ldl
/bin/sh ../libtool --silent  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../lib  -I../lib -I../trio -I/opt/zlib-1.2.8/include -I/usr/lib/include    -O2 -Wall -Wwrite-strings -Woverloaded-virtual -fno-exceptions -fno-rtti -fno-implement-inlines -MT Torrent.lo -MD -MP -MF .deps/Torrent.Tpo -c -o Torrent.lo Torrent.cc
In file included from ResMgr.h:29,
                 from FileAccess.h:29,
                 from Torrent.h:23,
                 from Torrent.cc:34:
xarray.h: In instantiation of `xarray_p<_xmap::entry>':
xmap.h:38:   instantiated from here
xarray.h:194: warning: `class xarray_p<_xmap::entry>' has virtual functions but
   non-virtual destructor
In file included from ArgV.h:24,
                 from FileAccess.h:31,
                 from Torrent.h:23,
                 from Torrent.cc:34:
xarray.h: In instantiation of `xarray_p<char>':
StringSet.h:27:   instantiated from `xarray_m<char>'
StringSet.h:27:   instantiated from here
xarray.h:194: warning: `class xarray_p<char>' has virtual functions but
   non-virtual destructor
xarray.h: In instantiation of `xarray_m<char>':
StringSet.h:27:   instantiated from here
xarray.h:233: warning: `class xarray_m<char>' has virtual functions but
   non-virtual destructor
In file included from Torrent.h:24,
                 from Torrent.cc:34:
xarray.h: In instantiation of `xarray_p<BeNode>':
Bencode.h:39:   instantiated from here
xarray.h:194: warning: `class xarray_p<BeNode>' has virtual functions but
   non-virtual destructor
In file included from Torrent.h:31,
                 from Torrent.cc:34:
xarray.h: In instantiation of `xarray_p<DHT::Peer>':
DHT.h:141:   instantiated from here
xarray.h:194: warning: `class xarray_p<DHT::Peer>' has virtual functions but
   non-virtual destructor
xarray.h: In instantiation of `xarray_p<xstring>':
xarray.h:241:   instantiated from `_xqueue<xstring*, xarray_p<xstring>, xstring*>'
DHT.h:167:   instantiated from `xqueue_p<xstring>'
DHT.h:167:   instantiated from here
xarray.h:194: warning: `class xarray_p<xstring>' has virtual functions but
   non-virtual destructor
In file included from Torrent.cc:34:
Torrent.h: In member function `void TorrentPiece::set_downloader(unsigned int,
   const TorrentPeer*, const TorrentPeer*, unsigned int)':
Torrent.h:116: variable-sized object of type `const TorrentPeer*[blk_count]'
   may not be initialized
In file included from TorrentTracker.h:23,
                 from Torrent.cc:35:
url.h: At global scope:
url.h:72: warning: `warn_unused_result' attribute directive ignored
make[1]: *** [Torrent.lo] Error 1
make[1]: Leaving directory `/root/lftp-4.6.3a/src'
make: *** [all-recursive] Error 1
```
Configure string: ./configure --prefix=/usr --with-openssl=/usr/lib --with-zlib=/opt/zlib-1.2.8
GCC: 
Reading specs from /usr/lib/gcc-lib/i386-redhat-linux/3.2.2/specs
Configured with: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --enable-shared --enable-threads=posix --disable-checking --with-system-zlib --enable-__cxa_atexit --host=i386-redhat-linux
Thread model: posix
gcc version 3.2.2 20030222 (Red Hat Linux 3.2.2-5)

Openssl 1.0.2d installed in /usr/lib

**Comment by areis422** [comment]
BUMP

**Comment by lavv17** [comment]
I have tried to fix the problem in bce803e, please try.
